saffron/src/blob.rs

```diff
@@ -161,4 +206,47 @@ mod tests {
             prop_assert_eq!(user_commitments, blob.commitments);
           }
         }
+
+    fn random_perturbation(threshold: f64, data: &[u8]) -> Vec<u8> {
+        let mut rng = rand::thread_rng();
+        data.iter()
+            .map(|b| {
+                let n = rng.gen::<f64>();
+                if n < threshold {
+                    rng.gen::<u8>()
+                } else {
+                    *b
+                }
+            })
+            .collect()
+    }
+
+    proptest! {
+    #![proptest_config(ProptestConfig::with_cases(10))]
+    #[test]
+        fn test_update((threshold, UserData(xs)) in (0.0..1.0f64).prop_flat_map(|t| {
+           UserData::arbitrary().prop_map(move |d| (t, d))
+        }))
+        {
+            // start with some random user data
+            let mut xs_blob = FieldBlob::<Vesta>::encode::<_, DefaultFqSponge<VestaParameters, PlonkSpongeConstantsKimchi>>(&*SRS, *DOMAIN, &xs);
```

Comment:
I guess you can also add a type alias for ` DefaultFqSponge<VestaParameters, PlonkSpongeConstantsKimchi>` at the top to get this cleaner.